Source/CarthageKit/Project.swift

```diff
@@ -187,11 +187,19 @@ public final class Project {
 		}
 
 		return cartfile.zipWith(privateCartfile)
-			.map { (var cartfile, privateCartfile) -> Cartfile in
-				cartfile.appendCartfile(privateCartfile)
+			.map { (var cartfile, privateCartfile) -> ColdSignal<Cartfile> in
+				// TODO: use Set when Carthage supports Swift 1.2
+				let duplicateDeps = filter(privateCartfile.dependencies) { privateDep in
+					contains(cartfile.dependencies, {$0.project == privateDep.project})
+				}
```

Comment:
As [hinted above](https://github.com/Carthage/Carthage/pull/364/files#r25580874), could we generalize this check to find any project that occurs more than once (whether the occurrences are split across `Cartfile` and `Cartfile.private`, or only in one)?